ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -79,162 +68,38 @@ export default class CompareSubtestDistributionView extends React.Component {
     });
   }
 
-  syncProjectGraphData = async (
-    projectName,
-    revision,
-    subtestSignature,
-    withBaseline,
-  ) => {
-    const {
-      graphSpecsField,
-      resultSetField,
-      replicateErrorField,
-      drawingDataField,
-      targetId,
-    } = withBaseline
-      ? {
-          graphSpecsField: 'graphOneSpecs',
-          resultSetField: 'originalResultSet',
-          replicateErrorField: 'originalReplicateError',
-          drawingDataField: 'baseDrawingData',
-          targetId: 'Base',
-        }
-      : {
-          graphSpecsField: 'graphTwoSpecs',
-          resultSetField: 'newResultSet',
-          replicateErrorField: 'newReplicateError',
-          drawingDataField: 'newDrawingData',
-          targetId: 'New',
-        };
-
-    const { replicateData, drawingData } = await this.fetchReplicateGraphData(
-      projectName,
-      revision,
-      subtestSignature,
-    );
-    const { resultSet, replicateDataError } = replicateData;
-    const numRuns = drawingData ? drawingData.numRuns : 0;
+  syncProjectResultSet = async (projectName, revision, resultSetField) => {
+    const { data } = await PushModel.getList({
+      repo: projectName,
+      commit_revision: revision,
+    });
+    const { results: pushPatch } = data;
+    const [first] = pushPatch;
```

Comment:
FYI - I'm working on a HOC for query param validation/resultset retrieval that we'll be able to use across all compare views so we'll be able to get rid of some of this that's repeated in each of the compare views. I'd like to turn it into a context once we switch to react-router so that if someone navigates from a different compare view, there isn't a need to fetch resultsets/projects/etc since it'll be shared in a "global" state. :)

Reply:
This sounds useful.